D/D.sublime-syntax

```diff
@@ -8,339 +8,1970 @@ file_extensions:
+  basic_type: 'bool|byte|cdouble|cent|cfloat|char|creal|dchar|double|float|idouble|ifloat|int|ireal|long|real|short|ubyte|ucent|uint|ulong|ushort|void|wchar|string|dstring|wstring'
+  language_constant: 'null|true|false|__FILE__|__FILE_FULL_PATH__|__MODULE__|__LINE__|__FUNCTION__|__PRETTY_FUNCTION__|__DATE__|__EOF__|__TIME__|__TIMESTAMP__|__VENDOR__|__VERSION__|__ctfe'
+  language_variable: 'this|super'
+  reserved: '{{keyword}}|{{basic_type}}|{{language_constant}}|{{language_variable}}'
+
+  operator_overloads: 'opNeg|opCom|opPostInc|opPostDec|opCast|opAdd|opSub|opSub_r|opMul|opDiv|opDiv_r|opMod|opMod_r|opAnd|opOr|opXor|opShl|opShl_r|opShr|opShr_r|opUShr|opUShr_r|opCat|opCat_r|opEquals|opEquals|opCmp|opCmp|opCmp|opCmp|opAddAssign|opSubAssign|opMulAssign|opDivAssign|opModAssign|opAndAssign|opOrAssign|opXorAssign|opShlAssign|opShrAssign|opUShrAssign|opCatAssign|opIndex|opIndexAssign|opCall|opSlice|opSliceAssign|opPos|opAdd_r|opMul_r|opAnd_r|opOr_r|opXor_r'
+
+  type_specialization_keyword: '__vector|function|delegate|super|return|__parameters'
+  type_specialization_storage: 'struct|union|class|interface|enum'
+
+  block_statement_loohahead: '(?={)'
+
 contexts:
+  prototype:
+    - include: comment-in
+
   main:
-    - match: /\*\*/
-      scope: comment.block.empty.d punctuation.definition.comment.d
-    - include: scope:text.html.javadoc
-    - match: |-
-        (?x)^\s*
-        ((?:\b(private|package|protected|public|export|static|final|native|synchronized|abstract)\b\s*)*) # modifier
-        (class|interface)\s+
-        (\w+)\s* # identifier
-        (?:\(\s*([^\)]+)\s*\)|)\s* # Template type
-        (?:
-          \s*(:)\s*
-          (\w+)
-          (?:\s*,\s*(\w+))?
-          (?:\s*,\s*(\w+))?
-          (?:\s*,\s*(\w+))?
-          (?:\s*,\s*(\w+))?
-          (?:\s*,\s*(\w+))?
-          (?:\s*,\s*(\w+))?
-        )? # super class
+    - meta_include_prototype: false
+    - match: '^#!.*$'
+      scope: comment.line.shebang.d
+      set: main-after-shebang-in
+    - match: '(?=.)'
+      set: main-after-shebang-in
+  main-after-shebang-in:
+    - include: attribute-specifier-in
+    - match: '\b(module)\s+({{identifier}})(;)'
+      scope: meta.module.d
       captures:
-        1: storage.modifier.d
-        3: storage.type.d
-        4: entity.name.type.class.d
-        5: storage.type.template.d
-        6: punctuation.separator.inheritance.d
-        7: entity.other.inherited-class.d
-        8: entity.other.inherited-class.d
-        9: entity.other.inherited-class.d
-        10: entity.other.inherited-class.d
-        11: entity.other.inherited-class.d
-        12: entity.other.inherited-class.d
-        13: entity.other.inherited-class.d
+        1: keyword.control.module.d
+        2: entity.name.namespace.d
+        3: punctuation.terminator.d
+      set: statement-list-in
+    - match: '\s'
+    - match: (?=.)
+      set: statement-list-in
```

Comment:
Instead of 78-80 just using `- match: (?=\S)` in 79 would be more suitable I think.